src/app.rs

```diff
@@ -1,255 +1,268 @@
 use ratatui::widgets::ListState;
+use tui_textarea::TextArea;
 
 use crate::{
-    history::{History, HistoryOperation},
-    tree::ElementTree,
+    entities::{TaskData, TaskEntity},
+    repository::AppTreeRepository,
 };
 
-const PLACEHOLDER_TEXT: &str = "Placeholder text.";
-const NEW_ELEMENT_TEXT: &str = "New element text.";
+const NEW_ELEMENT_TEXT: &str = "New task.";
+
+pub enum AppState {
+    NORMAL,
+    INSERT(u64),
```

Comment:
can you turn this tuple variant into a struct variant so we know what `u64` means just by looking at this (I saw later it's for id)